src/test/java/testsuite/integration/container/AuroraMySqlIntegrationTest.java

```diff
@@ -0,0 +1,596 @@
+    proxyMap.forEach((k, v) -> {
+      try {
+        v.toxics().get("DOWN-STREAM").remove();
+        v.toxics().get("UP-STREAM").remove();
+      } catch (Exception e) {
+        // Ignore as toxics were not set
+      }
+    });
+  }
+
+  @ParameterizedTest(name = "test_ConnectionString")
+  @MethodSource("generateConnectionString")
+  public void test_ConnectionString(String connStr, int port) throws SQLException {
+    final Connection conn = connectToInstance(connStr, port);
+    assertTrue(conn.isValid(5));
+    conn.close();
+  }
+
+  private static Stream<Arguments> generateConnectionString() {
+    return Stream.of(
+        Arguments.of(MYSQL_INSTANCE_1_URL, MYSQL_PORT),
+        Arguments.of(MYSQL_INSTANCE_1_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT),
+        Arguments.of(MYSQL_CLUSTER_URL, MYSQL_PORT),
+        Arguments.of(MYSQL_CLUSTER_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT),
+        Arguments.of(MYSQL_RO_CLUSTER_URL, MYSQL_PORT),
+        Arguments.of(MYSQL_RO_CLUSTER_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT)
+    );
+  }
+
+  @Test
+  public void test_ValidateConnectionWhenNetworkDown() throws SQLException, IOException {
+    final Connection conn = connectToInstance(MYSQL_INSTANCE_1_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT);
+    assertTrue(conn.isValid(5));
+
+    proxyInstance_1.toxics().bandwidth("DOWN-STREAM", ToxicDirection.DOWNSTREAM, 0); // from mysql server towards mysql driver
+    proxyInstance_1.toxics().bandwidth("UP-STREAM", ToxicDirection.UPSTREAM, 0); // from mysql driver towards mysql server
+
+    assertFalse(conn.isValid(5));
+
+    proxyInstance_1.toxics().get("DOWN-STREAM").remove();
+    proxyInstance_1.toxics().get("UP-STREAM").remove();
+
+    conn.close();
+  }
+
+  @Test
+  public void test_ConnectWhenNetworkDown() throws SQLException, IOException {
+    proxyInstance_1.toxics().bandwidth("DOWN-STREAM", ToxicDirection.DOWNSTREAM, 0); // from mysql server towards mysql driver
+    proxyInstance_1.toxics().bandwidth("UP-STREAM", ToxicDirection.UPSTREAM, 0); // from mysql driver towards mysql server
+
+    assertThrows(Exception.class, () -> {
+      // expected to fail since communication is cut
+      final Connection tmp = connectToInstance(MYSQL_INSTANCE_1_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT);
+    });
+
+    proxyInstance_1.toxics().get("DOWN-STREAM").remove();
+    proxyInstance_1.toxics().get("UP-STREAM").remove();
+
+    final Connection conn = connectToInstance(MYSQL_INSTANCE_1_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT);
+    conn.close();
+  }
+
+  @ParameterizedTest
+  @MethodSource("generateFailureDetectionTimeParams")
+  public void test_FailureDetectionTime(int detectionTime, int detectionInterval, int detectionCount, int sleepDelayMS) {
+    final Properties props = initDefaultProps();
+    props.remove(PropertyKey.connectTimeout.getKeyName());
+    props.remove(PropertyKey.socketTimeout.getKeyName());
```

Comment:
why are we removing these?

Reply:
To use the monitor's settings rather than the connections.